src/main/java/software/amazon/kinesis/connectors/flink/internals/publisher/fanout/FanOutShardSubscriber.java

```diff
@@ -215,7 +216,11 @@ private void handleError(final Throwable throwable) throws FanOutSubscriberExcep
 		LOG.warn("Error occurred on EFO subscription: {} - ({}).  {} ({})",
 			throwable.getClass().getName(), throwable.getMessage(), shardId, consumerArn, cause);
 
-		throw new FanOutSubscriberException(cause);
+		if (cause instanceof ReadTimeoutException) {
+			throw new RecoverableFanOutSubscriberException(cause);
```

Comment:
Can we provide some context for the future maintainers on why do we handle these exceptions differently? I'd make it clear that it's to avoid degradation under back-pressure situations. How about

> ReadTimeoutException occurs naturally under backpressure scenarios when full batches take longer to process than standard read timeout. Recoverable exceptions are intended to be retried indefinitely to avoid system degradation under backpressure.